modules_core/com.smf.jobs.defaults/src/com/smf/jobs/defaults/invoices/CreateFromOrder.java

```diff
@@ -108,16 +117,61 @@ protected ActionResult action(JSONObject parameters, MutableBoolean isStopped) {
       OBDal.getInstance().rollbackAndClose();
     }
 
+    ResponseActionsBuilder.MessageType messageType;
+    if (result.getType() == Result.Type.ERROR) {
+      messageType = ResponseActionsBuilder.MessageType.ERROR;
+    } else {
+      messageType = ResponseActionsBuilder.MessageType.SUCCESS;
+    }
+
     result.setResponseActionsBuilder(
         getResponseBuilder()
             .retryExecution()
             .refreshGridParameter(ORDER_GRID_PARAM)
-            .showMsgInProcessView(ResponseActionsBuilder.MessageType.SUCCESS, result.getMessage())
+            .showMsgInProcessView(messageType, result.getMessage())
     );
 
     return result;
   }
 
+  /**
+   * Checks if the BusinessPartner associated with an order specified in the given parameters has an invoice address.
+   *
+   * @param parameters
+   *     A JSONObject containing order details, including an order ID.
+   * @return true if the BusinessPartner associated with the order has an invoice address; false otherwise.
+   * @throws JSONException
+   *     If there is an error parsing the JSON data.
+   */
+  private boolean checkBusinessPartnerInvoiceAddress(JSONObject parameters) throws JSONException {
+    String orderId = parameters.getJSONObject(ORDER_GRID_PARAM).getJSONArray("_selection").getJSONObject(0).getString(
+        "id");
+
+    final OBCriteria<Order> orderOBCriteria = OBDal.getInstance().createCriteria(Order.class).add(
+        Restrictions.eq(Order.PROPERTY_ID, orderId));
+
+    Order order = (Order) orderOBCriteria.setMaxResults(1).uniqueResult();
+
+    return hasInvoiceAddress(order.getBusinessPartner());
+  }
+
+  /**
+   * Checks if the specified BusinessPartner has an invoice address.
+   *
+   * @param businessPartner
+   *     The BusinessPartner to check.
+   * @return true if the business partner has an invoice address; false otherwise.
+   */
+  private boolean hasInvoiceAddress(BusinessPartner businessPartner) {
+    final String hql = "select 1 from BusinessPartnerLocation bpl where bpl.invoiceToAddress = true and bpl.businessPartner = :bp";
```

Comment:
This function should be modified, although it works, it is not good practice to hardcode strings in the code, and from the BusinessPartner object it is possible to reach its BusinessPartnerLocation. Therefore, something like the following would be more optimal.

`private boolean hasInvoiceAddress(BusinessPartner businessPartner) {
    return businessPartner.getBusinessPartnerLocationList().stream()
            .anyMatch(BusinessPartnerLocation::isInvoiceToAddress);
}`